arangod/VocBase/vocbase.cpp

```diff
@@ -470,84 +461,54 @@ struct arangodb::VocBaseLogManager {
   Guarded<GuardedData> _guardedData;
 };
 
-/// @brief increase the reference counter for a database
-bool TRI_vocbase_t::use() {
-  auto expected = _refCount.load(std::memory_order_relaxed);
-  while (true) {
-    if ((expected & 1) != 0) {
-      // deleted bit is set
+bool TRI_vocbase_t::use() noexcept {
+  auto v = _refCount.load(std::memory_order_relaxed);
+  do {
+    if ((v & 1) != 0) {
       return false;
     }
-    // increase the reference counter by 2.
-    // this is because we use odd values to indicate that the database has been
-    // marked as deleted
-    auto updated = expected + 2;
-    TRI_ASSERT((updated & 1) == 0);
-    if (_refCount.compare_exchange_weak(expected, updated,
-                                        std::memory_order_release,
-                                        std::memory_order_relaxed)) {
-      // compare-exchange worked. we're done
-      return true;
-    }
-    // compare-exchange failed. try again!
-    expected = _refCount.load(std::memory_order_relaxed);
-  }
+  } while (!_refCount.compare_exchange_weak(v, v + 2, std::memory_order_acquire,
```

Comment:
Why is the memory order `acquire` here and not `release`?

Reply:
Because we need to see markAsDropped, and our +2 not needed for synchronization code.

You can look on intrusive_ptr as an a good example of this pattern
https://www.boost.org/doc/libs/1_81_0/libs/atomic/doc/html/atomic/usage_examples.html#boost_atomic.usage_examples.example_reference_counters

Reply:
For -2 we need release because we want to synchronize with isDangling